src/main/java/com/terraformersmc/terrestria/surface/ThreeLayerOutlinedSurfaceConfig.java

```diff
@@ -0,0 +1,47 @@
+package com.terraformersmc.terrestria.surface;
+
+import com.mojang.datafixers.Dynamic;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.world.gen.surfacebuilder.TernarySurfaceConfig;
+
+public class ThreeLayerOutlinedSurfaceConfig extends TernarySurfaceConfig {
+
+	private final BlockState middle;
+	private final float middleNoisePoint;
+	private final BlockState outer;
+	private final float outerNoisePoint;
+
+	public ThreeLayerOutlinedSurfaceConfig(BlockState inner, BlockState topMiddle, float middleNoisePoint, BlockState topEdge, float outerNoisePoint, BlockState under, BlockState underwater) {
```

Comment:
Maybe just take a TernarySurfaceConfig in the constructor? This would simplify `deserialize` as well.